crates/binjs_io/src/multipart/write.rs

```diff
@@ -107,22 +107,31 @@ struct TableEntry<T> where T: Clone + std::fmt::Debug { // We shouldn't need the
     /// The index, actually computed in `write()`.
     index: TableIndex<T>
 }
-impl<T> TableEntry<T> where T: Clone + std::fmt::Debug {
+impl<T> TableEntry<T> where T: Clone + std::fmt::Debug + Ord {
     fn new(data: T) -> Self {
         TableEntry {
             instances: RefCell::new(1),
             index: TableIndex::new(&format!("{:?}", data)),
             data,
         }
     }
+
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match u32::cmp(&*other.instances.borrow(), &*self.instances.borrow()) {
```

Comment:
Turns out that this pattern is so common that `Ordering` has a method `then_with` to factorize it :)

So, something along the lines of:

```rust
u32::cmp(...)
  .then_with(|| self.data.cmp(&other.data)
```